stake-pool/program/src/processor.rs

```diff
@@ -1410,37 +1435,136 @@ impl Processor {
-        )?;
+        let maybe_split_from_account_info =
+            if let Some(ephemeral_stake_seed) = maybe_ephemeral_stake_seed {
+                let ephemeral_stake_account_info = maybe_ephemeral_stake_account_info.unwrap();
+                let ephemeral_stake_bump_seed = check_ephemeral_stake_address(
+                    program_id,
+                    stake_pool_info.key,
+                    ephemeral_stake_account_info.key,
+                    ephemeral_stake_seed,
+                )?;
+                let ephemeral_stake_account_signer_seeds: &[&[_]] = &[
+                    EPHEMERAL_STAKE_SEED_PREFIX,
+                    &stake_pool_info.key.to_bytes(),
+                    &ephemeral_stake_seed.to_le_bytes(),
+                    &[ephemeral_stake_bump_seed],
+                ];
+                create_stake_account(
+                    ephemeral_stake_account_info.clone(),
+                    ephemeral_stake_account_signer_seeds,
+                    system_program_info.clone(),
+                )?;
 
-        // split into transient stake account
-        Self::stake_split(
-            stake_pool_info.key,
-            reserve_stake_account_info.clone(),
-            withdraw_authority_info.clone(),
-            AUTHORITY_WITHDRAW,
-            stake_pool.stake_withdraw_bump_seed,
-            total_lamports,
-            transient_stake_account_info.clone(),
-        )?;
+                // split into ephemeral stake account
+                Self::stake_split(
+                    stake_pool_info.key,
+                    reserve_stake_account_info.clone(),
+                    withdraw_authority_info.clone(),
+                    AUTHORITY_WITHDRAW,
+                    stake_pool.stake_withdraw_bump_seed,
+                    total_lamports,
+                    ephemeral_stake_account_info.clone(),
+                )?;
 
-        // activate transient stake to validator
-        Self::stake_delegate(
-            transient_stake_account_info.clone(),
-            validator_vote_account_info.clone(),
-            clock_info.clone(),
-            stake_history_info.clone(),
-            stake_config_info.clone(),
-            withdraw_authority_info.clone(),
-            stake_pool_info.key,
-            AUTHORITY_WITHDRAW,
-            stake_pool.stake_withdraw_bump_seed,
-        )?;
+                // activate stake to validator
+                Self::stake_delegate(
+                    ephemeral_stake_account_info.clone(),
+                    validator_vote_account_info.clone(),
+                    clock_info.clone(),
+                    stake_history_info.clone(),
+                    stake_config_info.clone(),
+                    withdraw_authority_info.clone(),
+                    stake_pool_info.key,
+                    AUTHORITY_WITHDRAW,
+                    stake_pool.stake_withdraw_bump_seed,
+                )?;
+                if validator_stake_info.transient_stake_lamports > 0 {
```

Comment:
could use this condition to select the split destination.  would dereplicode a bit